x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/filter/SecurityActionFilter.java

```diff
@@ -87,38 +84,25 @@ public SecurityActionFilter(AuthenticationService authcService, AuthorizationSer
             throw LicenseUtils.newComplianceException(XPackField.SECURITY);
```

Comment:
Is the functional requirement above to block certain APIs when security becomes disabled because the license expired still valid?

I think features that are available on higher license levels only ought to have a behavior when the license expires. I would think that an expired license grants the same features as the basic license, maybe something extra to not upset anyone but still nudge them to renew.
Because Security will be enabled by default even on the basic license, it doesn't feel right to "cripple" it when a license expires below the level one would get at no charge.
Granted the crippling behavior here is not really Security related.

So, I have two questions:
- Do we preserve the same "crippling" behavior in 8? I think @bytebilly is most in position to answer this.
- Do we want this behavior to be implemented in Security? I don't think so.

None of this have to hold this PR.

Reply:
Good catch @albertzaharovits ! I agree with you that the functionality you get in basic should not be different than what you get in an expired trial/gold/platinum. Can you please open an issue around this or bring it up for discussion in the next team meeting so that we don't lose track of it ? Thanks